bot/exts/evergreen/status_codes.py

```diff
@@ -20,62 +22,63 @@ class HTTPStatusCodes(commands.Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
-    @commands.group(name="http_status", aliases=("status", "httpstatus"), invoke_without_command=True)
+    @commands.group(
+        name="http_status",
+        aliases=("status", "httpstatus"),
+        invoke_without_command=True,
+    )
     async def http_status_group(self, ctx: commands.Context, code: int) -> None:
         """Choose a cat or dog randomly for the given status code."""
         subcmd = choice((self.http_cat, self.http_dog))
         await subcmd(ctx, code)
 
     @http_status_group.command(name="cat")
     async def http_cat(self, ctx: commands.Context, code: int) -> None:
-        """Sends an embed with an image of a cat, portraying the status code."""
-        embed = discord.Embed(title=f"**Status: {code}**")
-        url = HTTP_CAT_URL.format(code=code)
-
-        try:
-            HTTPStatus(code)
-            async with self.bot.http_session.get(url, allow_redirects=False) as response:
-                if response.status != 404:
-                    embed.set_image(url=url)
-                else:
-                    raise NotImplementedError
-
-        except ValueError:
-            embed.set_footer(text="Inputted status code does not exist.")
-
-        except NotImplementedError:
-            embed.set_footer(text="Inputted status code is not implemented by http.cat yet.")
-
-        finally:
-            await ctx.send(embed=embed)
+        """Send a cat version of the requested HTTP status code."""
+        if code in range(100, 599):
+            await self.build_embed(url=HTTP_CAT_URL.format(code=code), ctx=ctx, code=code)
+            return
+        await ctx.send(
+            embed=discord.Embed(
+                title="Input status code does not exist",
+                description="The range of accepted status codes that are ok is from 100 to 599",
```

Comment:
"accepted status codes that are ok" seems a bit verbose, "The range of valid status codes is 100 to 599" might be a bit more concise. Same would apply to the other one, you could probably create a function that takes the url and runs the command to get rid of the repetition here, although having `build_embed` does remove most of the repetition so that's ok

Reply:
I decided to create a standalone embed that would be sent in the case that the status code is outside of the range. I used your wording since it reads much easier.